trackbasket_be/models/krogerservice.py

```diff
@@ -46,7 +46,7 @@ def closest_store(cls, zipcode):
 
     blacklist = ['shell company']
     i = 0
-    while (parsed_response['data'][i]['name'].lower() in blacklist) and (i < number_results):
+    while (parsed_response['data'][i]['name'].lower() in blacklist) and (i < number_results - 1):
```

Comment:
since we're starting at index `0` we need to finish at index `number_results - 1`

Reply:
good catch here